mcs/class/Mono.Compiler/Mono.Compiler.BigStep/LLVM/BitCodeEmitter.cs

```diff
@@ -0,0 +1,612 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+using Mono.Compiler;
+using SimpleJit.Metadata;
+using SimpleJit.CIL;
+
+using LLVMSharp;
+
+/// <summary>
+///   Emit LLVM bitcode via LLVMSharp.
+/// </summary>
+namespace Mono.Compiler.BigStep.LLVMBackend
+{
+    public class BitCodeEmitter : IOperationProcessor
+    {
+        private static readonly LLVMBool Success = new LLVMBool(0);
+        private static readonly LLVMBool True = new LLVMBool(1);
+        private static readonly LLVMMCJITCompilerOptions s_options = new LLVMMCJITCompilerOptions { NoFramePointerElim = 0 };
+
+        private static int s_moduleSeq;
+        private static bool s_uninitialized;
+
+        private LLVMModuleRef module;
+        private LLVMBuilderRef builder;
+        private LLVMValueRef function;
+
+        private LLVMValueRef[] argAddrs;
+        private LLVMValueRef[] localAddrs;
+        private Dictionary<string, LLVMValueRef> temps;
```

Comment:
Longer-term I kind of wish these stringly-keyed dictionaries were strongly typed.